Server/mods/deathmatch/logic/CGame.cpp

```diff
@@ -1343,6 +1353,15 @@ void CGame::InitialDataStream(CPlayer& Player)
 
     marker.Set("onPlayerJoin");
 
+    SString joinSecret = Player.GetDiscordJoinSecret();
+    if (joinSecret.length())
+    {
+        CLuaArguments Arguments;
+        Arguments.PushBoolean(true);
+        Arguments.PushString(joinSecret);
+        Player.CallEvent("onPlayerDiscordJoin", Arguments);

```

Comment:
I'd maybe generalize this a bit to something like `onPlayerMessengerJoin` to be not too tightly coupled to Discord.

The signature may be something like: `string messengerType, bool justConnected, string secret`.

Reply:
Perhaps the third argument could be a table that changes depending on the "messenger"? Is "messenger" correct terminology here? Not sure.

Reply:
onPlayerExternalJoin maybe?